grammar.y.in

```diff
@@ -592,11 +593,41 @@ other:	  pqual TK_BROADCAST	{ CHECK_PTR_VAL(($$ = gen_broadcast(cstate, $1))); }
 	| PPPOES		{ CHECK_PTR_VAL(($$ = gen_pppoes(cstate, 0, 0))); }
 	| GENEVE pnum		{ CHECK_PTR_VAL(($$ = gen_geneve(cstate, $2, 1))); }
 	| GENEVE		{ CHECK_PTR_VAL(($$ = gen_geneve(cstate, 0, 0))); }
+	| BATADV pbatadv	{ $$ = $2; }
 	| pfvar			{ $$ = $1; }
 	| pqual p80211		{ $$ = $2; }
 	| pllc			{ $$ = $1; }
 	;
 
+pbatadv:		{ CHECK_PTR_VAL(($$ = gen_batadv(cstate, 0, 0, 0, 0))); }
+	| pnum		{ CHECK_PTR_VAL(($$ = gen_batadv(cstate, $1, 1, 0, 0))); }
+	| pnum pnum	{ CHECK_PTR_VAL(($$ = gen_batadv(cstate, $1, 1, $2, 1))); }
+	| pnum ID
+		{
+			int type;
+			CHECK_INT_VAL($1);
+			CHECK_PTR_VAL($2);
+
+			switch ($1) {
+			case 14:
+				type = pcap_nametobatadvtype_v14($2);
+				break;
+			case 15:
+				type = pcap_nametobatadvtype_v15($2);
+				break;
+			default:
+				YYABORT;
```

Comment:
This should probably call `bpf_set_error()` to report that $1 is an unsupported batman-adv version.

Reply:
Thanks! I've added an according error message with bpf_set_error().